api/api/serializers/media_serializers.py

```diff
@@ -653,7 +653,11 @@ def validate_source_field(value):
 
             allowed_sources = list(search_controller.get_sources(media_type).keys())
             sources = value.lower().split(",")
-            sources = [source for source in sources if source in allowed_sources]
+            original_len = len(sources)
+            valid_sources = [source for source in sources if source in allowed_sources]
+            if original_len > len(sources):
+                invalid_sources = set(sources).difference(set(valid_sources))
+                logger.warn(f"Invalid sources in search query: {invalid_sources}")
             value = ",".join(sources)
```

Comment:
```suggestion
            # Ensure there are no duplicates in this set
            valid_sources = set([source for source in sources if source in allowed_sources])
            if  len(sources) > len(valid_sources):
                invalid_sources = set(sources).difference(valid_sources)
                logger.warn(f"Invalid sources in search query: {invalid_sources}; sources query'{sources}'")
            value = ",".join(list(valid_sources))
```
@manojks1999 , sorry for late review. I was pretty sure I added one a couple of days ago, but I cannot find it now 🤯  The code I wrote in the issue wasn't fully functional.

In this suggestion, we first filter out all of the invalid sources, and also convert the sources to a set to make sure there are no duplicate values. If the `valid_sources` set contains more item than the original `sources` list, then we did filter out some sources, so we log the filtered out sources as a warning. We also log the original sources in case the filtered out values were duplicates (these will not show up as invalid sources).
In the end, we need to join the _valid_ sources and return that.